### PR TITLE
Exposed new APIs and tidied up existing implementation

### PR DESCRIPTION
**Pull request by itstheceo**
This pull request addresses the following improvements and issues:

# Justification

This pull request addresses the following improvements and issues, ultimately bringing the plugin closer to the available native counterparts.

In addition to removing unimplemented properties, the typings have been pruned, updated and documented. A concise interface remains that is platform agnostic unless where otherwise stated.

Both demos have been aligned and updated accordingly, Angular or otherwise.

## Added the `completionBlock` property

A block to be executed upon completion of the animation. The animation is considered complete when it finishes playing and is no longer looping. An example has been added showcasing this.

## Added the `playAnimationFromProgressToProgress` function

Plays the animation from the specified start and end progress values. This is particularly useful when a portion of your animation can be looped. A rudimentary example has been added showcasing this, in addition to a more advanced example showcasing how it can be used in conjunction with the `completionBlock` property to seamlessly transition between portions of a composition after some async task has finished.

## Added the `setColorValueDelegateForKeyPath` function

Sets the provided color value on each property that matches the specified keyPath. An example has been added showcasing this.

## Added the `setOpacityValueDelegateForKeyPath` function

Sets the provided opacity value on each property that matches the specified keyPath. An example has been added showcasing this.

## Removed the `theme` property

Lottie has no notion of this construct. The existing dynamic color functionality can be achieved with the `setColorValueDelegateForKeyPath` function. We shouldn't be imposing unnecessary constructs on the consumer.

## Removed the LottieHelper jar

NativeScript is able to marshall the required types on it's own, and as such this helper is not required.

## Aligned the Android and iOS implementations

There was lots of unnecessary code in the iOS implementation in particular, because the native view was being inflated inside an additional subview for some reason.

### CODE REVIEW

**Comment by Codex-** [comment]
I suggest a major version bump for these changes

File: src/nativescript-lottie.android.ts [hunk 1]
```diff
@@ -189,3 +242,7 @@ export enum CacheStrategy {
   Weak = com.airbnb.lottie.LottieAnimationView.CacheStrategy.Weak,
   Strong = com.airbnb.lottie.LottieAnimationView.CacheStrategy.Strong
 }
+
+function clamp(val: number, min: number = 0, max: number = 1) {
```
Comment (Codex-) [inline src/nativescript-lottie.android.ts]: Clamp is copied twice, could be abstracted to a commons and imported/consumed.

**Comment by hamdiwanis** [comment]
hey @Codex- 
"NativeScript is able to marshall the required types on its own, and as such, this helper is not required"
good luck with it when the LottieValueCallback required generics. that is why a helper was needed.
"We shouldn't be imposing unnecessary constructs on the consumer"
how is it imposing? it's not like you HAVE to use it :D. also, don't see what is the problem of having a declarative API to theme Lottie element instead of having to reference it every time to do it. 
anyway good work 👍

**Comment by itstheceo** [comment]
@hamdiwanis 
No luck is needed, the Helper is not required. Java generics are implemented via type erasure which removes all information regarding generic types during compilation; adding another pre-compiled jar isn't going to change this.

**Comment by hamdiwanis** [comment]
@itstheceo i think you missed the point , if lottie callback required generics and nativescript does not support it then it will not find any native method with the same signature to call. then how would you deal with this?

**Comment by itstheceo** [comment]
@hamdiwanis I think your point misses the point, but feel free to continue telling me the Helper is required when this branch works without it.

**Comment by hamdiwanis** [comment]
@itstheceoI see you are not conversation friendly :D so let's end it here and have a nice day :)

**Comment by itstheceo** [comment]
@hamdiwanis I'm more than happy to have a discussion about it, but generics have nothing to do with it as I said the first time. I urge you to pull the branch and see for yourself before claiming otherwise.

File: src/nativescript-lottie.android.ts (hunk 1 above)
Comment (itstheceo) [reply src/nativescript-lottie.android.ts]: Nice catch, fixed in https://github.com/bradmartin/nativescript-lottie/pull/40/commits/166b84b9b0338209308258875e3d15cd0588d6af

**Comment by dnoiz1** [comment]
+1 to make this branch mainline release.

**Comment by bradmartin** [comment]
I'll try to get to this early next week. 

Thanks for all the work @itstheceo 

@hamdiwanis - maybe we can provide an alias on the `theme` as well. I haven't used all the plugin offers in some time so I'll have to get up to speed on using `theme` vs. what this PR implements. If it's something that can be aliased tho relatively simple to provide both as options for consumers of the plugin without a lot of overhead then I'm all for keeping it as an option.

**Comment by dnoiz1** [comment]
@hamdiwanis I would be fine with an external API change that causes older versions to break as long as this is reflected in versioning. This should take care of people relying on the current API who already have working apps.

**Comment by bradmartin** [comment]
I see both sides here. As long as everything is documented on the README with the breaking changes, we'll bump major and leave it at that. 

I like the idea of `theme` but I also like mapping everything to the native counterparts as close as possible, just makes it somewhat easier to reason about in the future with maintenance and updates.

@hamdiwanis - has done a great job with this plugin in the past and this PR by @itstheceo is also great 🎉 . Thanks everyone for the hard work.

I'll likely get to the merging and publishing Monday or Tuesday this week.

**Comment by Codex-** [comment]
May not have published correctly?
https://www.npmjs.com/package/nativescript-lottie
`"Unable to find a readme for nativescript-lottie@3.0.0"`

**Pull request merged**